packages/noya-state/src/reducers/pointReducer.ts

```diff
@@ -172,6 +182,168 @@ export function pointReducer(
         });
       });
     }
+    case 'setControlPointX':
+    case 'setControlPointY': {
+      if (!state.selectedControlPoint) return state;
+      const [type, amount, CanvasKit, mode] = action;
+      const controlPointType = state.selectedControlPoint.controlPointType;
+      const axis = type === 'setControlPointX' ? 'x' : 'y';
+
+      const pageIndex = getCurrentPageIndex(state);
+      const layerIndexPaths = getSelectedLayerIndexPaths(state);
+      const boundingRects = getBoundingRectMap(
+        getCurrentPage(state),
+        [state.selectedControlPoint.layerId],
+        {
+          clickThroughGroups: true,
+          includeArtboardLayers: false,
+          includeHiddenLayers: false,
+        },
+      );
+
+      return produce(state, (draft) => {
+        layerIndexPaths.forEach((indexPath) => {
+          const page = draft.sketch.pages[pageIndex];
+          const layer = Layers.access(page, indexPath);
+          const pointList = draft.selectedControlPoint?.pointIndex;
+          const curveMode = draft.selectedControlPoint?.curveMode;
+          const boundingRect = boundingRects[layer.do_objectID];
+
+          if (!Layers.isPointsLayer(layer) || !boundingRect) return;
+
+          // Update all points by first transforming to the canvas's coordinate system
+          layer.points
+            .filter((_, index) => pointList === index)
+            .forEach((curvePoint) => {
```

Comment:
Can we make this a `map` instead of a `forEach`? It's usually easier to maintain code that doesn't mutate state, and in this case we have to map the points later to encode them anyway so there's no immer-related perf gain here